p24/src/refs.rs

```diff
@@ -1,29 +1,36 @@
 // f1 returns first or second element of tuple depending on the flag.
 pub fn f1(pair: &mut (u32, u32), flag: bool) -> &mut u32 {
     match flag {
-        true => &mut pair.0,
-        false => &mut pair.1,
+        true => &mut pair.1,
+        false => &mut pair.0,
     }
 }
 
 // f2 returns n-th element of slice.
 pub fn f2(slice: &mut [u32], n: usize) -> &mut u32 {
+    if n >= slice.len() {
+        panic!("Index out of bounds");
+    }
```

Comment:
This check may be removed, since you will get a panic in `slice[n]` and there is not opportunity for overflow/underflow.